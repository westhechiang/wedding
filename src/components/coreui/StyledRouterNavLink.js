import styled from 'styled-components';
import { NavLink } from 'react-router-dom';
import { theme } from '../../ui';

const types = {
  noPad: `
    padding: 0;
    min-width: 0;
  `,
  undefined: `
    padding: 0 10px;
  `,
};

const hoverStyles = {
  tertiaryNav: `
    border-bottom: 3px ${theme.color.blue} solid;

    p {
      color: ${theme.color.blue4};
    }
  `,
  primaryNav: `
    border-bottom: 3px ${theme.color.cyan4} solid;

    p {
      color: ${theme.color.white};
    }
  `,
  borderBottomFinancialsSidebar: `
    border-bottom: 1px ${theme.color.white} solid;
  `,
};

const activeStyles = {
  tertiaryNav: `
    border-bottom: 3px ${theme.color.blue} solid;

    p {
      color: ${theme.color.blue4};
    }
  `,
  borderBottomFinancialsSidebar: `
    border-bottom: 1px ${theme.color.white} solid;
  `,
  primaryNav: `
    border-bottom: 3px ${theme.color.cyan4} solid;
  `,
  darkBlueBackground: `
    background-color: ${theme.color.blueDark};
  `,
};

const StyledRouterNavLink = styled(NavLink)`
  text-decoration: ${props => props.textDecoration || 'none'};
  height: ${props => (props.height || '85px')};
  width: ${props => (props.width || 'auto')};
  color: ${theme.color.cyan4};
  min-width: 50px;
  display: block;
  ${props => types[props.type]}

  &.active {
    ${props => activeStyles[props['data-activestyle']]}
  }

  &:hover {
    ${props => hoverStyles[props['data-hoverstyle']]}
  }
`;

export default StyledRouterNavLink;
