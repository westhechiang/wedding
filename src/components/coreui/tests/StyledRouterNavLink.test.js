import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import StyledRouterNavLink from '../../../components/coreui/StyledRouterNavLink';

describe('<StyledRouterNavLink />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<StyledRouterNavLink to="/" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its snapshot when passed a "height" prop', () => {
    const wrapper = shallow(<StyledRouterNavLink to="/" height="50px" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
