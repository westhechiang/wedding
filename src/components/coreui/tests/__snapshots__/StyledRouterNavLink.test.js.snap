// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledRouterNavLink /> should match its empty snapshot 1`] = `
<NavLink
  activeClassName="active"
  className="sc-bdVaJa hciKHX"
  to="/"
/>
`;

exports[`<StyledRouterNavLink /> should match its snapshot when passed a "height" prop 1`] = `
<NavLink
  activeClassName="active"
  className="sc-bdVaJa dpbkTe"
  height="50px"
  to="/"
/>
`;
